tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/server/AbstractRenameTest.java

```diff
@@ -0,0 +1,328 @@
+		final List<RenamePosition> positions;
+		/** The new name used during rename. */
+		final String newName;
+		/** Test expectation for the source code after the rename happened. */
+		final Map<String, Map<String, String>> projectsModulesExpectedSourcesAfter;
+
+		/** Creates a new {@link RenameTestConfiguration}. */
+		public RenameTestConfiguration(
+				Map<String, Map<String, String>> projectsModulesSourcesBefore,
+				List<RenamePosition> positions,
+				String newName,
+				Map<String, Map<String, String>> projectsModulesExpectedSourcesAfter) {
+
+			this.projectsModulesSourcesBefore = projectsModulesSourcesBefore;
+			this.positions = ImmutableList.copyOf(positions);
+			this.newName = newName;
+			this.projectsModulesExpectedSourcesAfter = projectsModulesExpectedSourcesAfter;
+		}
+	}
+
+	/** Position where a rename should be initiated. */
+	public static class RenamePosition {
+		/** Name of the project containing the module in which the rename should be initiated. */
+		final String projectName;
+		/** Name of the module in which the rename should be initiated. */
+		final String moduleName;
+		/** Line of the cursor position where the rename should be initiated. */
+		final int line;
+		/** Column of the cursor position where the rename should be initiated. */
+		final int column;
+
+		/** Creates a new {@link RenamePosition}. */
+		public RenamePosition(String projectName, String moduleName, int line, int column) {
+			this.projectName = projectName;
+			this.moduleName = moduleName;
+			this.line = line;
+			this.column = column;
+		}
+	}
+
+	/** Call this method in a single-file test. */
+	protected void testAtCursors(CharSequence sourceBefore, String newName, CharSequence expectedSourceAfter) {
+		Pair<String, String> sourceBeforeAsPair = Pair.of(
+				TestWorkspaceManager.DEFAULT_MODULE_NAME,
+				sourceBefore.toString());
+		Pair<String, String> expectedSourceAfterAsPair = Pair.of(
+				TestWorkspaceManager.DEFAULT_MODULE_NAME,
+				expectedSourceAfter.toString());
+		testAtCursors(
+				Collections.singletonList(sourceBeforeAsPair),
+				newName,
+				Collections.singletonList(expectedSourceAfterAsPair));
+	}
+
+	/** Call this method in a single-project, multi-file test. */
+	protected void testAtCursors(
+			Iterable<Pair<String, String>> modulesSourcesBefore,
+			String newName,
+			Iterable<Pair<String, String>> modulesExpectedSourcesAfter) {
+
+		Pair<String, ? extends Iterable<Pair<String, String>>> sourceBeforeAsPair = Pair.of(
+				TestWorkspaceManager.DEFAULT_PROJECT_NAME,
+				modulesSourcesBefore);
+		Pair<String, ? extends Iterable<Pair<String, String>>> expectedSourceAfterAsPair = Pair.of(
+				TestWorkspaceManager.DEFAULT_PROJECT_NAME,
+				modulesExpectedSourcesAfter);
+		testAtCursorsWS(
+				Collections.singletonList(sourceBeforeAsPair),
+				newName,
+				Collections.singletonList(expectedSourceAfterAsPair));
+	}
+
+	/** Call this method in a multi-project test. */
+	protected void testAtCursorsWS(
```

Comment:
Consider to move up the features regarding multi-cursor tests and default dependencies to n4js-runtime, since these two features could be re-used by other tests as well.